worker/src/features/batchExport/handleBatchExportJob.ts

```diff
@@ -16,12 +19,193 @@ import {
+            Prisma.sql`
+            s.id,
+            s."created_at" AS "createdAt",
+            s.bookmarked,
+            s.public,
+            t."userIds",
+            t."countTraces",
+            o."sessionDuration",
+            o."inputCost" AS "inputCost",
+            o."outputCost" AS "outputCost",
+            o."totalCost" AS "totalCost",
+            o."promptTokens" AS "inputTokens",
+            o."completionTokens" AS "outputTokens",
+            o."totalTokens" AS "totalTokens",
+            t."tags" AS "traceTags",
+            (count(*) OVER ())::int AS "totalCount" 
+          `,
+            {
+              projectId,
+              filter: filter
+                ? [...filter, createdAtCutoffFilter]
+                : [createdAtCutoffFilter],
+              orderBy,
+              limit: pageSize,
+              page: Math.floor(offset / pageSize),
+            }
+          );
+          const chunk = await prisma.$queryRaw<BatchExportSessionsRow[]>(query);
+
+          return chunk;
+        },
+        1000,
+        env.BATCH_EXPORT_ROW_LIMIT
+      );
+    case "traces": {
+      const {
+        orderByCondition,
+        filterCondition,
+        scoreTimestampFilterCondition,
+      } = parseTracesOrderByAndFilter(
+        orderBy,
+        filter ? [...filter, createdAtCutoffFilter] : [createdAtCutoffFilter]
+      );
+
+      const distinctScoreNames = await prisma.$queryRaw<{ name: string }[]>`
+        SELECT DISTINCT name
+        FROM scores s
+        WHERE s.project_id = ${projectId}
+        AND s.created_at <= ${cutoffCreatedAt}
+        ${scoreTimestampFilterCondition}
+      `;
+
+      const emptyScoreColumns = distinctScoreNames.reduce(
+        (acc, { name }) => ({ ...acc, [name]: null }),
+        {}
+      );
+
+      return new DatabaseReadStream<unknown>(
+        async (pageSize: number, offset: number) => {
+          const query = createTracesQuery({
+            select: Prisma.sql`
+              t."bookmarked",
+              t."id",
+              t."timestamp",
+              t."name",
+              t."user_id" AS "userId",
+              tm."level" AS "level",
+              tl."observationCount" AS "observationCount",
+              s_avg."scores_values" AS "scores",
+              tl.latency AS "latency",
+              t."release",
+              t."version",
+              t.session_id AS "sessionId",
+              t."input",
+              t."output",
+              t."metadata",
+              t."tags",
+              COALESCE(tm."promptTokens", 0)::bigint AS "usage.promptTokens",
+              COALESCE(tm."completionTokens", 0)::bigint AS "usage.completionTokens",
+              COALESCE(tm."totalTokens", 0)::bigint AS "usage.totalTokens",
+              COALESCE(tm."calculatedInputCost", 0)::numeric AS "inputCost",
+              COALESCE(tm."calculatedOutputCost", 0)::numeric AS "outputCost",
+              COALESCE(tm."calculatedTotalCost", 0)::numeric AS "totalCost"
```

Comment:
_Question_: I have ensured all possible trace table rows are included in our query here. However for the sessions query we have included a `totalCount` column which is not included in the table, may I ask why it was included in the first place or was it legacy? And would we like to keep it? Also asking as it's quite expensive to compute.